### PR TITLE
Add helpers to create IPC command buffer headers and descriptors

### PR DESCRIPTION
**Pull request by yuriks**
This should hopefully encourage people to add these (without using magic numbers) to service returns in the future.

### CODE REVIEW

**Comment by yuriks** [comment]
I was additionally thinking of moving `GetCommandBuffer` to the `Ipc` namespace too, but that would make ripple changes through all the service functions, and I'm not sure what people think of it.

**Comment by Subv** [comment]
LGTM

**Comment by archshift** [comment]
LGTM

File: src/core/hle/kernel/session.h [hunk 1]
```diff
@@ -8,6 +8,40 @@
 #include "core/hle/kernel/thread.h"
 #include "core/memory.h"
 
+namespace Ipc {
```
Comment (neobrain) [inline src/core/hle/kernel/session.h]: *IPC ?

**Comment by neobrain** [comment]
LGTM other than my comment.

File: src/core/hle/kernel/session.h [hunk 2]
```diff
@@ -8,6 +8,40 @@
...
+}
+
+inline u32 StaticBufferDesc(u32 size, unsigned int buffer_id) {
+    return 0x2 | (size << 14) | ((buffer_id & 0xF) << 10);
+}
+
+enum MappedBufferPermissions {
+    R = 2,
+    W = 4,
+    RW = 6,
```
Comment (linkmauve) [inline src/core/hle/kernel/session.h]: Replace 6 with R | W, which describes better what it is.

**Comment by linkmauve** [comment]
Other than that, LGTM.

**Comment by Subv** [comment]
@mailwl I think it's better to be explicit in this case, lest the person using it just forgets about it

**Comment by yuriks** [comment]
Yeah, I did consider default parameters, but it's indeed better to make it explicit.

**Comment by linkmauve** [comment]
Btw, I’ve been thinking about making every service call return a std::pair with the header and the return code, and to have the wrapper do the actual assignment to cmd_buffer[0] and cmd_buffer[1], how practical would that be?

**Comment by yuriks** [comment]
I think that should be done using a wrapper for reading/writing to the command buffer, not by tacking it onto the IPC system, since, for example, the return code is just a convention used by most of the calls, it's not a requirement of the IPC system. Doing that would just further distance the emulation from how the console works.

**Comment by yuriks** [comment]
Updated

**Pull request merged**